client/my-sites/guided-transfer/host-credentials-page/index.jsx

```diff
@@ -29,16 +40,34 @@ class HostCredentialsPage extends Component {
 		this.setFieldValue( fieldName, e.target.value );
 	}
 
+	redirectToCart = () => {
+		upgradesActions.addItem( cartItems.guidedTransferItem() );
+		page.redirect( `/checkout/${ this.props.siteSlug }` );
+	}
+
 	submit = () => {
-		// Future PR: This function will trigger the Redux API call
+		const payload = {
+			...this.state.fieldValues,
+			host_slug: this.props.hostSlug
+		};
+
+		this.props.submit( payload )
+			.then( didSubmit => {
```

Comment:
Don't depend on an action creator's promise resolution, as this breaks [the unidirectional data flow contract](http://redux.js.org/docs/basics/DataFlow.html).